llvm/lib/Target/AArch64/AArch64FrameLowering.cpp

```diff
@@ -3355,8 +3360,8 @@ bool AArch64FrameLowering::spillCalleeSavedRegisters(
           MachineMemOperand::MOStore, Size, Alignment));
       MIB.addReg(PnReg);
       MIB.addReg(AArch64::SP)
-          .addImm(RPI.Offset) // [sp, #offset*scale],
-                              // where factor*scale is implicit
+          .addImm(RPI.Offset / 2) // [sp, #offset*2*scale],
+                                  // where scale is implicit
```

Comment:
Yeah I was debating this comment. It depends what does the "offset" part in the comment mean. If it refers to RPI.Offset, then current comment is accurate, if it refers to RPI.Offset/2 then this change is correct. I thought that because RPI.Offset has "offset" in the name, I would  make "offset" in the comment refer to it.

Reply:
I read 'offset' as meaning the actual encoded immediate `RPI.Offset / 2`, because (along with the SP reg that is added on the line above), that is the MachineInstr that is being built here. The semantics of the encoded address is then `[sp, #offset * 2 * vscale]`, where `2 * vscale` is implied by the instruction, which is what I think the comment tries to clarify.

What about replacing `#offset` by `#imm`, such that we end up with:
```
// [sp, #imm*2*vscale]
// where 2*vscale is implicit
```
?